<a href="https://colab.research.google.com/github/PosgradoMNA/actividades-de-aprendizaje-JersonDavidPerezContreras/blob/main/Actividad_Semanal_6%2C_visualizaci%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



# TECNOLOGICO DE MONTEREY





#Maestria en inteligencia artificial aplicada





##Ciencia y Analitica de datos





## Profesora:

María de la Paz Rico Fernández





# Presentado por: 

Jerson David Perez Contreras

A01793810



In [2]:
# lIBRERIAS

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import RobustScaler, MinMaxScaler, StandardScaler
from sklearn.decomposition import PCA

##1. Descarga los datosEnlaces a un sitio externo. y carga el dataset en tu libreta.


df = pd.read_csv('https://raw.githubusercontent.com/PosgradoMNA/Actividades_Aprendizaje-/main/default%20of%20credit%20card%20clients.csv')
col_names = ['ID', 'Amount_Credit', 'Gender', 'Education', 'Marital_status', 'Age', 'Repayment_status_September', 'Repayment_status_August', 
       'Repayment_status_July', 'Repayment_status_June', 'Repayment_status_May', 'Repayment_status_April','Amount_bill_statement_September_2005',
        'Amount_bill_statement_August_2005', 'Amount_bill_statement_July_2005', 'Amount_bill_statement_June_2005', 'Amount_bill_statement_May_2005',
        'Amount_bill_statement_April_2005','Amount_paid_September_2005', 'Amount_paid_August_2005', 'Amount_paid_July_2005', 'Amount_paid_June_2005', 
        'Amount_paid_May_2005','Amount_paid_April_2005', 'y']
df.columns = col_names

df.head()

ID  Amount_Credit  Gender  Education  Marital_status   Age  \
0   1          20000     2.0        2.0             1.0  24.0   
1   2         120000     2.0        2.0             2.0  26.0   
2   3          90000     2.0        2.0             2.0  34.0   
3   4          50000     2.0        2.0             1.0  37.0   
4   5          50000     1.0        2.0             1.0  57.0   

   Repayment_status_September  Repayment_status_August  Repayment_status_July  \
0                         2.0                      2.0                   -1.0   
1                        -1.0                      2.0                    0.0   
2                         0.0                      0.0                    0.0   
3                         0.0                      0.0                    0.0   
4                        -1.0                      0.0                   -1.0   

   Repayment_status_June  ...  Amount_bill_statement_June_2005  \
0                   -1.0  ...                              0.0   
1                    0.0  ...                           3272.0   
2                    0.0  ...                          14331.0   
3                    0.0  ...                          28314.0   
4                    0.0  ...                          20940.0   

   Amount_bill_statement_May_2005  Amount_bill_statement_April_2005  \
0                             0.0                               0.0   
1                          3455.0                            3261.0   
2                         14948.0                           15549.0   
3                         28959.0                           29547.0   
4                         19146.0                           19131.0   

   Amount_paid_September_2005  Amount_paid_August_2005  Amount_paid_July_2005  \
0                         0.0                    689.0                    0.0   
1                         0.0                   1000.0                 1000.0   
2                      1518.0                   1500.0                 1000.0   
3                      2000.0                   2019.0                 1200.0   
4                      2000.0                  36681.0                10000.0   

   Amount_paid_June_2005  Amount_paid_May_2005  Amount_paid_April_2005    y  
0                    0.0                   0.0                     0.0  1.0  
1                 1000.0                   0.0                  2000.0  1.0  
2                 1000.0                1000.0                  5000.0  0.0  
3                 1100.0                1069.0                  1000.0  0.0  
4                 9000.0                 689.0                   679.0  0.0  

[5 rows x 25 columns]

In [3]:
## 2.  Obten la información del DataFrame con los métodos y propiedades: shape, columns, head(), dtypes, info(), isna()

print( "Tamaño de Data: ")
print(df.shape) ## tiene 30000 observaciones con 25 variables
print( "\n")
print( "Columnas: ")
print( df.columns)

print( "\n")
print( "Encabezado Datos: ")
print( df.head())

Tamaño de Data: 
(30000, 25)


Columnas: 
Index(['ID', 'Amount_Credit', 'Gender', 'Education', 'Marital_status', 'Age',
       'Repayment_status_September', 'Repayment_status_August',
       'Repayment_status_July', 'Repayment_status_June',
       'Repayment_status_May', 'Repayment_status_April',
       'Amount_bill_statement_September_2005',
       'Amount_bill_statement_August_2005', 'Amount_bill_statement_July_2005',
       'Amount_bill_statement_June_2005', 'Amount_bill_statement_May_2005',
       'Amount_bill_statement_April_2005', 'Amount_paid_September_2005',
       'Amount_paid_August_2005', 'Amount_paid_July_2005',
       'Amount_paid_June_2005', 'Amount_paid_May_2005',
       'Amount_paid_April_2005', 'y'],
      dtype='object')


Encabezado Datos: 
   ID  Amount_Credit  Gender  Education  Marital_status   Age  \
0   1          20000     2.0        2.0             1.0  24.0   
1   2         120000     2.0        2.0             2.0  26.0   
2   3          90000     2.0       

In [4]:
print( "Tipos de variables: ")
print(df.dtypes)


Tipos de variables: 
ID                                        int64
Amount_Credit                             int64
Gender                                  float64
Education                               float64
Marital_status                          float64
Age                                     float64
Repayment_status_September              float64
Repayment_status_August                 float64
Repayment_status_July                   float64
Repayment_status_June                   float64
Repayment_status_May                    float64
Repayment_status_April                  float64
Amount_bill_statement_September_2005    float64
Amount_bill_statement_August_2005       float64
Amount_bill_statement_July_2005         float64
Amount_bill_statement_June_2005         float64
Amount_bill_statement_May_2005          float64
Amount_bill_statement_April_2005        float64
Amount_paid_September_2005              float64
Amount_paid_August_2005                 float64
Amount_paid_July_20

In [5]:
print( "Detalle de variables: ")

df.info()

Detalle de variables: 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 25 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   ID                                    30000 non-null  int64  
 1   Amount_Credit                         30000 non-null  int64  
 2   Gender                                29999 non-null  float64
 3   Education                             29998 non-null  float64
 4   Marital_status                        29998 non-null  float64
 5   Age                                   29995 non-null  float64
 6   Repayment_status_September            29997 non-null  float64
 7   Repayment_status_August               29995 non-null  float64
 8   Repayment_status_July                 29993 non-null  float64
 9   Repayment_status_June                 29991 non-null  float64
 10  Repayment_status_May                  29984 non-null  float

In [6]:
print( "Contiene nulos?: ")

df.isna().any()  ## PODEMOS OBSERVAR QUE SI EXISTEB VALORES  NA EN LAS DIFERENTES CAMPOS

Contiene nulos?: 


ID                                      False
Amount_Credit                           False
Gender                                   True
Education                                True
Marital_status                           True
Age                                      True
Repayment_status_September               True
Repayment_status_August                  True
Repayment_status_July                    True
Repayment_status_June                    True
Repayment_status_May                     True
Repayment_status_April                   True
Amount_bill_statement_September_2005     True
Amount_bill_statement_August_2005        True
Amount_bill_statement_July_2005          True
Amount_bill_statement_June_2005          True
Amount_bill_statement_May_2005           True
Amount_bill_statement_April_2005         True
Amount_paid_September_2005               True
Amount_paid_August_2005                  True
Amount_paid_July_2005                    True
Amount_paid_June_2005             

In [13]:
## 3.  Limpia los datos eliminando los registros nulos o rellena con la media de la columna


(df.isna().sum()/df.shape[0])*100## Observar cuantos datos hay nulos

ID                                      0.000000
Amount_Credit                           0.000000
Gender                                  0.003333
Education                               0.006667
Marital_status                          0.006667
Age                                     0.016667
Repayment_status_September              0.010000
Repayment_status_August                 0.016667
Repayment_status_July                   0.023333
Repayment_status_June                   0.030000
Repayment_status_May                    0.053333
Repayment_status_April                  0.046667
Amount_bill_statement_September_2005    0.036667
Amount_bill_statement_August_2005       0.036667
Amount_bill_statement_July_2005         0.043333
Amount_bill_statement_June_2005         0.050000
Amount_bill_statement_May_2005          0.056667
Amount_bill_statement_April_2005        0.033333
Amount_paid_September_2005              0.026667
Amount_paid_August_2005                 0.030000
Amount_paid_July_200

In [14]:
#Los datos  nulos no pesan mas del 1% por tal motivo se escoge la eliminacion de estoas datos
df.dropna(inplace=True)
(df.isna().sum()/df.shape[0])*100  ## Observar cuantos datos hay nulos

ID                                      0.0
Amount_Credit                           0.0
Gender                                  0.0
Education                               0.0
Marital_status                          0.0
Age                                     0.0
Repayment_status_September              0.0
Repayment_status_August                 0.0
Repayment_status_July                   0.0
Repayment_status_June                   0.0
Repayment_status_May                    0.0
Repayment_status_April                  0.0
Amount_bill_statement_September_2005    0.0
Amount_bill_statement_August_2005       0.0
Amount_bill_statement_July_2005         0.0
Amount_bill_statement_June_2005         0.0
Amount_bill_statement_May_2005          0.0
Amount_bill_statement_April_2005        0.0
Amount_paid_September_2005              0.0
Amount_paid_August_2005                 0.0
Amount_paid_July_2005                   0.0
Amount_paid_June_2005                   0.0
Amount_paid_May_2005            

In [15]:
## Ya no existes datos nulos o vacios

## 4. Calcula la estadística descriptiva con describe() y explica las medidas de tendencia central y dispersión

df.describe()

ID   Amount_Credit        Gender     Education  \
count  29958.000000    29958.000000  29958.000000  29958.000000   
mean   15005.550504   167555.900928      1.604012      1.853094   
std     8654.547473   129737.299088      0.489070      0.790471   
min        1.000000    10000.000000      1.000000      0.000000   
25%     7516.250000    50000.000000      1.000000      1.000000   
50%    15005.500000   140000.000000      2.000000      2.000000   
75%    22497.750000   240000.000000      2.000000      2.000000   
max    30000.000000  1000000.000000      2.000000      6.000000   

       Marital_status           Age  Repayment_status_September  \
count    29958.000000  29958.000000                29958.000000   
mean         1.551739     35.483443                   -0.017124   
std          0.521952      9.214319                    1.123989   
min          0.000000     21.000000                   -2.000000   
25%          1.000000     28.000000                   -1.000000   
50%          2.000000     34.000000                    0.000000   
75%          2.000000     41.000000                    0.000000   
max          3.000000     79.000000                    8.000000   

       Repayment_status_August  Repayment_status_July  Repayment_status_June  \
count             29958.000000           29958.000000           29958.000000   
mean                 -0.134021              -0.166767              -0.221110   
std                   1.197171               1.196026               1.168419   
min                  -2.000000              -2.000000              -2.000000   
25%                  -1.000000              -1.000000              -1.000000   
50%                   0.000000               0.000000               0.000000   
75%                   0.000000               0.000000               0.000000   
max                   8.000000               8.000000               8.000000   

       ...  Amount_bill_statement_June_2005  Amount_bill_statement_May_2005  \
count  ...                     29958.000000                    29958.000000   
mean   ...                     43279.335370                    40328.984578   
std    ...                     64364.684347                    60826.219326   
min    ...                   -170000.000000                   -81334.000000   
25%    ...                      2327.500000                     1762.250000   
50%    ...                     19037.500000                    18104.500000   
75%    ...                     54551.250000                    50220.750000   
max    ...                    891586.000000                   927171.000000   

       Amount_bill_statement_April_2005  Amount_paid_September_2005  \
count                      29958.000000                29958.000000   
mean                       38889.925763                 5664.614460   
std                        59582.883301                16568.823518   
min                      -339603.000000                    0.000000   
25%                         1256.000000                 1000.000000   
50%                        17067.500000                 2100.000000   
75%                        49234.750000                 5007.000000   
max                       961664.000000               873552.000000   

       Amount_paid_August_2005  Amount_paid_July_2005  Amount_paid_June_2005  \
count             2.995800e+04           29958.000000           29958.000000   
mean              5.925715e+03            5228.429969            4829.873556   
std               2.305598e+04           17617.338167           15676.205514   
min               0.000000e+00               0.000000               0.000000   
25%               8.352500e+02             390.000000             296.250000   
50%               2.009000e+03            1800.000000            1500.000000   
75%               5.000000e+03            4511.500000            4014.750000   
max               1.684259e+06          896040.000000          621000.000000   

       Amount_p

Describe nos sirve para mirar un detalle estadisticos de cada columna, en esta momento se observa que todas las columnas tiene 29958 observacione, esto debido a que ya se realizo una limpieza de datos. Entre los datos descritivos podemos encontrar el promedio, su desviación, el minimo y el maximo y los percentiles 25,50 y 75.
Estos valores nos permite conocer la naturaleza de los datos y su comportamiento, por ejemplo la edad tiene un rango entre los 21 y 79 años, con un promedio de 35 años, Tambien vemos como las variables categoricas no presentan datos estadisticos muy legibles, pero en el caso del genero se puede concluir que mas del 50% son mujeres al estar presentes en el persentil 50 y 75 y su media pasa por encima de 1,5 tendiendo a 2.


In [ ]:
## Realiza el conteo de las variables categóricas

